server/src/main/java/io/crate/planner/optimizer/costs/PlanStats.java

```diff
@@ -136,41 +138,44 @@ public Stats visitUnion(Union union, Void context) {
         }
 
         @Override
-        public Stats visitNestedLoopJoin(NestedLoopJoin join, Void context) {
+        public Stats visitJoinPlan(JoinPlan join, Void context) {
+            return visitAbstractJoinPlan(join, context);
+        }
+
+        @Override
+        public Stats visitHashJoin(HashJoin join, Void context) {
+            return visitAbstractJoinPlan(join, context);
+        }
+
+        public Stats visitAbstractJoinPlan(AbstractJoinPlan join, Void context) {
             var lhsStats = join.lhs().accept(this, context);
             var rhsStats = join.rhs().accept(this, context);
-            Map<ColumnIdent, ColumnStats<?>> statsByColumn = Maps.concat(lhsStats.statsByColumn(), rhsStats.statsByColumn());
+            Map<ColumnIdent, ColumnStats<?>> statsByColumn = Maps.concat(lhsStats.statsByColumn(),
+                                                                         rhsStats.statsByColumn());
             if (lhsStats.numDocs() == -1
                 || lhsStats.sizeInBytes() == -1
                 || rhsStats.numDocs() == -1
                 || rhsStats.sizeInBytes() == -1) {
                 return new Stats(-1, -1, statsByColumn);
             }
-            long numRows = join.joinType() == JoinType.CROSS
-                ? lhsStats.numDocs() * rhsStats.numDocs()
-                // We don't have any cardinality estimates, so just take the bigger table
-                : Math.max(lhsStats.numDocs(), rhsStats.numDocs());
-            Stats joinStats = new Stats(
-                numRows,
-                (lhsStats.averageSizePerRowInBytes() * numRows) + (rhsStats.averageSizePerRowInBytes() * numRows),
-                statsByColumn
-            );
-            Symbol joinCondition = join.joinCondition();
-            if (joinCondition == null) {
-                return joinStats;
-            }
+            long numRows = Math.max(lhsStats.numDocs(), rhsStats.numDocs());
+            long sizeInBytes =
+                (numRows * lhsStats.averageSizePerRowInBytes())
+                + (numRows * rhsStats.averageSizePerRowInBytes());
+
+            Stats joinStats = new Stats(numRows, sizeInBytes, statsByColumn);
```

Comment:
Shouldn't this still take the joinType into consideration?

Reply:
Good catch, this was by mistake.